invenio_rdm_records/marshmallow/json.py

```diff
@@ -95,34 +95,15 @@ class CreatorSchemaV1(BaseSchema):
 class ContributorSchemaV1(CreatorSchemaV1):
     """Contributor schema."""
 
-    ROLES = [
-        "ContactPerson",
-        "DataCollector",
-        "DataCurator",
-        "DataManager",
-        "Distributor",
-        "Editor",
-        "HostingInstitution",
-        "Producer",
-        "ProjectLeader",
-        "ProjectManager",
-        "ProjectMember",
-        "RegistrationAgency",
-        "RegistrationAuthority",
-        "RelatedPerson",
-        "Researcher",
-        "ResearchGroup",
-        "RightsHolder",
-        "Sponsor",
-        "Supervisor",
-        "WorkPackageLeader",
-        "Other"
-    ]
+    role = SanitizedUnicode(required=True)
 
-    role = SanitizedUnicode(required=True, validate=validate.OneOf(
-                choices=ROLES,
-                error=_('Invalid role. {input} not one of {choices}.')
-            ))
+    @validates_schema
+    def validate_data(self, data, **kwargs):
+        """Validate role."""
+        vocabulary = Vocabularies.get_vocabulary('contributors.role')
```

Comment:
Yes, definitely passed the 3 strike-refactor mark. I made a `validate_entry` function in json.py, so that the Vocabularies class is isn't tied or responsible for marshmallow specific functionality.